### PR TITLE
mvp

### PR DESCRIPTION
**Pull request by mileszander**

### CODE REVIEW

File: client/src/components/App.jsx [hunk 1]
```diff
@@ -1,15 +1,30 @@
...
 class App extends React.Component {
   constructor () {
     super();
     this.state = {
-      id: 0
+      id: 0,
+      reviews: []
     }
-    //bind functions here
+    this.componentDidMount = this.componentDidMount.bind(this)
```
Comment (rcalimlim) [inline client/src/components/App.jsx:16]: only need this if you're passing it down to another component

File: db/config.js [hunk 2]
```diff
@@ -4,7 +4,66 @@ mongoose.connect('mongodb://localhost/test', {useNewUrlParser: true});
...
+
+
+
+
+const getAll = (callback) => {
+  db.collection("test").find({}).toArray((err, result) => {
+    if (err) {
+      callback(err)
+    } else {
+      callback(null, result)
```
Comment (rcalimlim) [inline db/config.js:22]: Queries like `find` in mongoose/mongodb return a [`Query`](https://mongoosejs.com/docs/queries.html#queries-are-not-promises) instance which is thenable like Promises. Just a heads up if you don't like using the callback pattern.

File: db/config.js [hunk 3]
```diff
@@ -4,7 +4,66 @@ mongoose.connect('mongodb://localhost/test', {useNewUrlParser: true});
...
+//   dbo.collection("customers").find({}).toArray(function(err, result) {
+//     if (err) throw err;
+//     console.log(result);
+//     db.close();
+//   });
+// });
+
+
+module.exports = {
+  getAll : getAll,
```
Comment (rcalimlim) [inline db/config.js:42]: A common pattern I've been seeing is the[ property shorthand functionality in ES6](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Property_definitions). You can just list the property name if your key/value pair are the same.
`module.exports = { getAll };`

File: server/server.js [hunk 4]
```diff
@@ -16,10 +17,17 @@ app.use(express.static('client/dist'));
...
-  res.status(200).send('Get request successful!');
-});
+app.get('/test', function (req, res) {
+  db.getAll((err, result) => {
+    if (err) {
+      res.send(`couldn't get`)
+      console.log(`we didn't get it`)
+    } else {
+      res.send(result)
+      console.log('we got it')
```
Comment (rcalimlim) [inline server/server.js:27]: Refactoring with Promises might be cleaner here.

**Review by rcalimlim — CHANGES_REQUESTED** [review]
Looks good--I made some comments related to refactoring with Promises (instead of straight callbacks) for readability and an ES6 object literal notation.

File: db/config.js (hunk 3 above)
Comment (mileszander) [reply db/config.js:42]: updated

File: server/server.js (hunk 4 above)
Comment (mileszander) [reply server/server.js:27]: gonna stick with cb's till I get the hang of promises

**Pull request merged**